skpro/distributions/histogram.py

```diff
@@ -0,0 +1,145 @@
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+"""Histogram distribution."""
+
+__author__ = ["ShreeshaM07"]
+
+import numpy as np
+
+from skpro.distributions.base import BaseDistribution
+
+
+class Histogram(BaseDistribution):
+    """Histogram Probability Distribution.
+
+    The histogram probability distribution is parameterized
+    by the bins and bin densities.
+
+    Parameters
+    ----------
+    bins : float or array of float 1D
+        array has the bin boundaries with 1st element the first bin's
+        starting point and rest are the bin ending points of all bins
+    bin_mass: array of float 1D
+        Mass of the bins or Area of the bins.
+        Sum of all the bin_mass must be 1.
+    index : pd.Index, optional, default = RangeIndex
+    columns : pd.Index, optional, default = RangeIndex
+    """
+
+    def __init__(self, bins, bin_mass, index=None, columns=None):
+        self.bins = bins
+        self.bin_mass = bin_mass
+
+        super().__init__(index=index, columns=columns)
+
+    def _mean(self):
+        """Return expected value of the distribution.
+
+        Returns
+        -------
+        float, sum(bin_mass)/range(bins)
+            expected value of distribution (entry-wise)
+        """
+        bins = self.bins
+        # 1 is the cumulative sum of all bin_mass
+        return 1 / (max(bins) - min(bins))
```

Comment:
that's not correct? Also, you need to be careful about the different cases of `bins` being int or iterable.

Reply:
I will take care of the `bins` cases. But in the case where `bins` has the bin edges then shouldn't this be the mean as mean = `sum(bin_width*bin_height)/sum(bin_width)`,the numerator is basically area under the histogram which is = `1` and the sum of bin_width would be the range of the `bins` values thus = `max(bins)- min(bins)`. Is that incorrect?

Reply:
Could you please review if what I have considered for the `mean` and `var` is correct?

Reply:
I think your formula is simply incorrect.
The correct formula for mean is:

Let $b_0, \dots, b_n$ the bin boundaries, and $m_i, 1= 1,\dots, n$ the mass in the bin $[b_{i-1}, b_i]$.

The mean of the histogram distirbution is then

$\mu = \sum_{i=1}^n (b_i - b{i-1}) p_i$ - which you can obtain by applying `np.dot` and `np.diff`.

Reply:
for "easy" computation of the mean and variance, you can use that the histogram distribution is the same as the two-step conditional where you first sample which bin you are in, with probabilities $m_i$, and then from the uniform within the bin.

Use the conditional formulae for mean and variance on this idea - this also shows why the mean has the above form, as the weighted mean of means of uniform distributions corresponding to uniforms on the bins.

Reply:
Oh yes that is correct, I've made a mistake I will correct it now. Thanks for the help.